opentelemetry-exporter-gcp-trace/src/opentelemetry/exporter/cloud_trace/__init__.py

```diff
@@ -357,24 +363,38 @@ def _strip_characters(ot_version):
 }
 
 
-def _extract_resources(resource: Resource) -> Dict[str, str]:
+def _extract_resources(
+    resource: Resource, resource_regex: Optional[Pattern] = None
+) -> Dict[str, str]:
+    extracted_attributes = {}
     resource_attributes = resource.attributes
+    if resource_regex:
+        extracted_attributes.update(
+            {
+                k: str(v)
+                for k, v in resource_attributes.items()
+                if resource_regex.match(k)
+            }
+        )
     if resource_attributes.get("cloud.provider") != "gcp":
-        return {}
+        return extracted_attributes
     resource_type = resource_attributes["gcp.resource_type"]
     if (
         not isinstance(resource_type, str)
         or resource_type not in OT_RESOURCE_ATTRIBUTE_TO_GCP
     ):
-        return {}
-    return {
-        "g.co/r/{}/{}".format(resource_type, gcp_resource_key,): str(
-            resource_attributes[ot_resource_key]
-        )
-        for ot_resource_key, gcp_resource_key in OT_RESOURCE_ATTRIBUTE_TO_GCP[
-            resource_type
-        ].items()
-    }
+        return extracted_attributes
+    extracted_attributes.update(
+        {
+            "g.co/r/{}/{}".format(resource_type, gcp_resource_key,)
+            if resource_type == "gce_instance"
+            else ot_resource_key: str(resource_attributes[ot_resource_key])
```

Comment:
I don't think this if/else is needed, we still special case both keys of `OT_RESOURCE_ATTRIBUTE_TO_GCP`.

Reply:
ok, I misread you previous comment.  changed this back.